### PR TITLE
Limit max amount of messages in consensus queue

### LINKED ISSUE #1117: Implement a max size for the consensus queue
This should prevent OOM errors as we've seen them on the devnet.

### PR DESCRIPTION
**Pull request by julesdesmit**
Fixes #1117

### CODE REVIEW

File: pkg/core/consensus/agreement/step.go [hunk 1]
```diff
@@ -105,7 +105,10 @@ func (s *Loop) shouldCollectNow(a message.Message, round uint64, queue *consensu
 		return false
 	}
 
-	if hdr.Round > round {
+	// Only store events up to 10 rounds into the future
+	// XXX: According to protocol specs, we should abandon consensus
+	// if we notice valid messages from far into the future.
+	if hdr.Round > round && hdr.Round-round < 10 {
```
Comment (goshawk-3) [inline pkg/core/consensus/agreement/step.go:111]: Out of curiosity, have we noticed a devnet node collecting an Agreement message from  so far round? 
Much earlier, we should have this node in `OutSync` mode and Consensus/Agreement goroutines terminated.

File: pkg/core/consensus/queue.go [hunk 2]
```diff
@@ -68,13 +86,15 @@ func (eq *Queue) Clear(round uint64) {
...
 	newEntries := make(map[uint64]map[uint8][]message.Message)
+	eq.items = 0
 
 	for r := range eq.entries {
 		if r > round {
 			newEntries[r] = make(map[uint8][]message.Message)
 
 			for s, m := range eq.entries[r] {
 				newEntries[r][s] = m
+				eq.items++
```
Comment (goshawk-3) [inline pkg/core/consensus/queue.go:97]: Maybe this should be `eq.items += len(m)` ?

**Review by goshawk-3 — CHANGES_REQUESTED** [review]

File: pkg/core/consensus/queue.go (hunk 2 above)
Comment (julesdesmit) [reply pkg/core/consensus/queue.go]: Good catch!

File: pkg/core/consensus/agreement/step.go (hunk 1 above)
Comment (julesdesmit) [reply pkg/core/consensus/agreement/step.go:111]: Not yet, but this is possible in syncing/real network scenarios. Of course, there needs to be some proper guarding in place against maliciousness but it's part of the protocol to terminate and sync when noticing consensus messages from far in the future.

**Review by goshawk-3 — APPROVED** [review]
LGTM

**Pull request merged**